eventrecorder/recorder.go

```diff
@@ -165,19 +88,79 @@ func (r *EventRecorder) handleRetrievalEvents(res http.ResponseWriter, req *http
 			}
 		}
 	}
-	logger.Infow("Successfully submitted batch event insertion")
+
+	// Execute the batch
+	batchResult := r.db.SendBatch(ctx, &batchQuery)
+	err := batchResult.Close()
+	if err != nil {
+		totalLogger.Errorw("At least one retrieval event insertion failed", "err", err)
+		return err
+	}
+	totalLogger.Info("Successfully submitted batch event insertion")
+
+	return nil
 }
 
-func (r *EventRecorder) handleReady(res http.ResponseWriter, req *http.Request) {
-	switch req.Method {
-	case http.MethodGet:
-		// TODO: ping DB as part of readiness check?
-		res.Header().Add("Allow", http.MethodGet)
-	default:
-		http.Error(res, "", http.StatusMethodNotAllowed)
+func (r *EventRecorder) RecordAggregateEvents(ctx context.Context, events []AggregateEvent) error {
```

Comment:
Mmm, sure. We had talked about the new event structure replacing the old one entirely, so I was viewing this has a replacement. Since the http server was pulled out, the only thing really left was the DB pool and the different insert queries. Since it's the same DB and not two separate DBs, it made sense to keep the same pool and just add a new query. I don't mind pulling it out though.